chorus_lib/src/core.rs

```diff
@@ -205,23 +213,23 @@ impl<L1: ChoreographyLocation, B: Transport> Projector<L1, B> {
     /// Unwraps a located value at the projection target.
     ///
     /// Use this method to access the located value returned by a choreography.
-    pub fn unwrap<V: ChoreographicValue>(&self, located: &Located<V, L1>) -> V {
-        located.value.clone().unwrap()
+    pub fn unwrap<V>(&self, located: Located<V, L1>) -> V {
+        located.value.unwrap()
     }
 
     /// Performs end-point projection and runs a choreography.
-    pub fn epp_and_run<'a, V, C: Choreography<V>>(&'a self, choreo: &C) -> V {
+    pub fn epp_and_run<'a, V, C: Choreography<V>>(&'a self, choreo: C) -> V {
```

Comment:
All methods that `execute` a choreography now takes ownership of a choreography instance.